Core v2 docs update (#819)
spatialos/gdk-for-unity: docs/content/gameobject/interact-spatialos-monobehaviours.md

The SpatialOS GDK for Unity automatically injects the correct values into the these fields
as soon as the worker that this GameObject belongs to fulfills all the requirements.

## MonoBehaviour lifecycle

The GDK manages the lifecycle of all MonoBehaviours that contain at least
one field with a `[Require]` attribute. It ensures that these MonoBehaviours are disabled for

  * all Prefabs stored inside the `Resources` folder of your Unity project
  * all MonoBehaviours on instantiated GameObjects that have at least one requirement not fulfilled during runtime

> Whenever a MonoBehaviour is disabled, all fields decorated with the `[Require]` attribute are `null` and you can not use them as soon as `OnDisable` is called on that MonoBehaviour.

Reviewer: ```suggestion
Whenever a MonoBehaviour is disabled, all fields decorated with the `[Require]` attribute are `null` and you cannot use them as soon as `OnDisable` is called on that MonoBehaviour.
```
